scripts/scenario_01_waves.lua

```diff
@@ -191,9 +191,66 @@ function update(delta)
 	end
 	enemy_count = 0
 	friendly_count = 0
+	local order = nil
+	local order_target = nil
+	local ship_type = nil
 	for _, enemy in ipairs(enemyList) do
 		if enemy:isValid() then
 			enemy_count = enemy_count + 1
+			order = enemy:getOrder()
+			if order == "Defend Target" or order == "Attack" then
+				order_target = enemy:getOrderTarget()
+				if order_target ~= nil then
+					if not order_target:isValid() then
+						if random(1,100) < 50 then
+							enemy:orderRoaming()
+						else
+							enemy:orderAttack(getPlayerShip(-1))
+						end
+					end
+				else
+					if random(1,100) < 50 then
+						enemy:orderRoaming()
+					else
+						endmy:orderAttack(getPlayerShip(-1))
```

Comment:
I'm not a fan of a direct attack command, on multi ship games, this makes it ignore the 2nd ship. I noticed this when playing borderline fever online, some enemy ships would just blindly fly by players to attack a far away station. Making for easy kills.

Reply:
Instead, you could order a move towards the players current position, or even a "orderRoamingAt" order to roam towards the players current position.

Reply:
`orderDefendLocation` works very well for me when enemies should attack something without ignoring their sourrounding. `orderDefendTarget(player)` should work the same when the target is an enemy of the `CpuShip` and move.

As daid wrote, `orderAttack` is just a bit fanatic at times and makes it easy to kite enemies in a multiplayer scenario.

Reply:
Removed `orderAttack` in favor of `orderFlyTowards` one of the human navy friendly stations as an option when the enemy gets stuck in `orderIdle` apparently when the target of `orderDefend` goes bad